salt/utils/network.py

```diff
@@ -2153,6 +2153,18 @@ def refresh_dns():
         # Exception raised loading the library, thus RES_INIT is not defined
         pass
 
+def dns_resolve_addresses(host):
+    '''
+    This function returns all the IP address that the host looks up to.
+    IPv6 addresses are only returned if the python interpreter has been
+    compiled with IPv6 support, because otherwise the returned structure
+    is effectively garbage.
+    '''
```

Comment:
Thanks for your PR, @gczuczy . You're off to a good start here, but we probably need more information in order to make this documentation more effective.

Earlier this year, I started working on a set of standards for module documentation but had to put the project on hold temporarily. However, looking at what my research says should go into effective documentation for functions, it looks like the documentation would be more helpful if you could include:

- In the description, provide at least one use case. You should provide more than one if this function is particularly complex (such as a lot of kwargs), but it doesn't look like it is to me.
- Indicate the datatype for each parameter (string, dict, etc.). Returns should be included in the parameters.
- Provide an example of return data.

Your warning about how the IPv6 addresses need to be compiled with an interpreter that has IPv6 support is good, but should probably be formatted as an admonition (a warning or tip box) in order to make it more obvious. Here's some more information about how to format warnings and tip boxes in rST: https://saltstack.gitlab.io/open/docs/docs-hub/topics/rst-guide.html#admonitions-tips-note-boxes-warnings

Reply:
@barbaricyawps Should the return type be specified by PEP484? I don't see that anywhere in the code currently.

Reply:
@barbaricyawps  Please see `8ce2431ace` for the docstring update, I hope this loves up to the upcoming standards.